File: mymtntest.js
describe('MTN App Launch Flow', function () {
    it('should launch app and verify number', function () {
        browser.setTimeout({ implicit: 30000 });

        // Click permission allow button
        const firstAction = $('id=com.android.permissioncontroller:id/permission_allow_button');
        firstAction.click();

        // Click get started button
        const secondAction = $('//android.view.ViewGroup[@content-desc="get_started_btn"]/android.view.ViewGroup');
        secondAction.click();

        // Input phone number
        const inputField = $('//android.widget.EditText[@content-desc="login_input_msisdn"]');
        inputField.setValue('0593130530');

        // Click checkbox
        const checkbox = $('android.widget.CheckBox');
        checkbox.click();

        // Click verify number button
        const verifyBtn = $('//android.view.ViewGroup[@content-desc="login_btn_verify_number"]/android.view.ViewGroup');
        verifyBtn.click();
    });
});

File: mymtn_test.py
from appium import webdriver
from appium.options.android import UiAutomator2Options

# Set up desired capabilities and pass the Android app-activity and app-package to Appium
options = UiAutomator2Options()
options.platformName = "Android"
options.deviceName = "1B271FDF6007B8"
options.automationName = "UiAutomator2"
options.appPackage = "com.mtngh.mymtn"
#options.appActivity = "com.mtngh.mymtn.MainActivity"
options.autoGrantPermissions = True
options.app = "C:\Users\owusu\Desktop\mtnapp.apk"
#options.noReset = True
#.ensureWebviewsHavePages = True

# Initialize the driver
app = webdriver.Remote("http://localhost:4723", options=options)

# Implicit wait for 30 seconds
app.implicitly_wait(30)

# Find the first action button and click on it
first_action = app.find_element("id", "com.android.permissioncontroller:id/permission_allow_button")
first_action.click()

# Find the second action button and click on it
second_action = app.find_element("xpath", '//android.view.ViewGroup[@content-desc="get_started_btn"]/android.view.ViewGroup')
second_action.click()

inputing_number = app.find_element("xpath", '//android.widget.EditText[@content-desc="login_input_msisdn"]')
inputing_number.send_keys("0593130530")

checkbox = app.find_element("class name", "android.widget.CheckBox")
checkbox.click()

verify_number = app.find_element("xpath", '//android.view.ViewGroup[@content-desc="login_btn_verify_number"]/android.view.ViewGroup')
verify_number.click()

#End the session
app.quit()

